File: Project1/project/c/block.c
//
// Created by linuxlite on 06-04-2019.
//

#include "block.h"

void Block(block *b, int bid, unsigned int x, unsigned int y){
    b->width = 1;
    b->height = 1;
    b->bid = bid;
    b->x = x;
    b->y = y;
}

void makeHorizontal(block *b){
    b->width=2;
    b->height=1;
}

void makeVertical(block *b){
    b->width=1;
    b->height=2;
}

void makeSquare(block *b){
    b->height=2;
    b->width=2;
}

unsigned int getX(block *b){ return b->x;}

unsigned int getY(block *b){ return b->y;}

unsigned int getWidth(block *b){ return b->width;}

unsigned int getHeight(block *b){ return b->height;}

bool moveUp(block *b){
    if(b->x > 0){
        b->x--;
        return true;
    } else
        return false;
}

bool moveDown(block *b){
    if(b->x < 4){
        b->x++;
        return true;
    } else
        return false;
}

bool moveLeft(block *b){
    if(b->y > 0){
        b->y--;
        return true;
    } else
        return false;
}

bool moveRight(block *b){
    if(b->y < 3){
        b->y++;
        return true;
    } else
        return false;
}

int getBid(block *b){ return b->bid;}

block copy(block *b){
    block ret;
    ret.bid = b->bid;
    ret.x = b->x;
    ret.y = b->y;
    ret.height = b->height;
    ret.width = b->width;
    return ret;
}

File: Project1/project/python/klotski/block.py
class Block:
    bid = -1
    x = -1
    y = -1
    height = 0
    width = 0

    def __init__(self, bid, x, y):
        if bid is 0:
            self.bid = 0
            self.x = 0
            self.y = 1
            self.width = 2
            self.height = 2

        else:
            self.bid = bid
            if 0 <= x < 5:
                self.x = x
            else:
                print(f'Wrong X value in block {bid}')
                return
            if 0 <= y < 4:
                self.y = y
            else:
                print(f'Wrong Y value in block {bid}')
                return
            self.height = 1
            self.width = 1

    def makeHorizontal(self):
        self.width = 2
        self.height = 1

    def makeVertical(self):
        self.height = 2
        self.width = 1

    def makeSquare(self):
        self.width = 1
        self.height = 1

    def moveUp(self):
        if self.y > 0:
            self.y -= 1
            return 1
        else:
            return 0

    def getX(self):
        return self.x

    def getY(self):
        return self.y

    def getWidth(self):
        return self.width

    def getHeight(self):
        return self.height

    def getCoords(self):
        coords = [self.x, self.y]
        return coords

    def moveUp(self):
        self.x -= 1

    def moveDown(self):
        self.x += 1

    def moveLeft(self):
        self.y -= 1

    def moveRight(self):
        self.y += 1

    def getBid(self):
        return self.bid

    def copy(self):
        newone = Block(int(self.bid), int(self.x), int(self.y))
        newone.height = int(self.height)
        newone.width = int(self.width)
        return newone
